tests/test_multistore_file.py

```diff
@@ -0,0 +1,40 @@
+"""Unit tests for oauth2client.multistore_file."""
+
+import errno
+import os
+import tempfile
+import unittest
+
+from oauth2client import multistore_file
+
+
+class _MockLockedFile(object):
+
+    def __init__(self, filename_str, error_code):
+        self.filename_str = filename_str
+        self.error_code = error_code
+        self.open_and_lock_called = False
+
+    def open_and_lock(self):
+        self.open_and_lock_called = True
+        raise IOError(self.error_code, '')
+
+    def is_locked(self):
+        return False
+
+    def filename(self):
+        return self.filename_str
+
+
+class MultistoreFileTests(unittest.TestCase):
+
+    def test_lock_file_raises_ioerror(self):
+        filehandle, filename = tempfile.mkstemp()
+        os.close(filehandle)
```

Comment:
Since we're not using the filehandle, there's no reason to leave it open (consuming a filehandle resource in the OS).

We need to use [tempfile.mkstemp()](https://docs.python.org/2/library/tempfile.html) to ensure that the temporary file (which the _Multistore constructor will open for reading via its name) is visible on the filesystem.  We're also responsible for cleaning the file up.  So removing the os.unlink() call was a mistake.

os.unlink and os.remove are identical, so that's purely stylistic.